mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +240,72 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _retryable_adls_function(self, func, artifact_path, **kwargs):
```

Comment:
To avoid the potential repeated calls to `_get_write_credential_infos`, we could have this method return the final credentials used and the calling points would overwrite their credentials with what is returned.  I am not a fan of this because it is tricky/clever code and tricky/clever code takes awhile to understand when other engineers have to read it.